migration/src/main/java/io/crate/migration/MigrationTool.java

```diff
@@ -0,0 +1,270 @@
+                                    }
+
+                                    if (!configuration.isDryrun()) {
+                                        LOGGER.debug("Migrating shard [{}] in directory [{}] for {}table [{}] " +
+                                                     "of node [{}]",
+                                            i,
+                                            shardIndexPath,
+                                            table.isPartitioned() ? "partitioned " : "",
+                                            tableName,
+                                            node);
+                                        new IndexUpgrader(shardDir, InfoStream.NO_OUTPUT, true).upgrade();
+                                        LOGGER.debug("Shard [{}] in directory [{}] for {}table [{}] of node [{}] " +
+                                                     "migrated successfully",
+                                            i,
+                                            shardIndexPath,
+                                            table.isPartitioned() ? "partitioned " : "",
+                                            tableName,
+                                            node);
+                                    }
+                                    if (!configuration.isDryrun()) {
+                                        summaryStats.addSuccessFull(table, node);
+                                    }
+                                }
+                            } catch (Exception e) {
+                                LOGGER.error("Error while upgrading {}table [{}] of node [{}]",
+                                    table.isPartitioned() ? "partitioned " : "",
+                                    tableName,
+                                    node,
+                                    e);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (!configuration.isDryrun()) {
+                LOGGER.info(summaryStats.print());
+            }
+        }
+    }
+
+    @VisibleForTesting
+    static void retrieveIndexDirs(Map<Table, List<File>> tableDirs, String path, List<String> tableNames) {
+        File[] directories = new File(path).listFiles(File::isDirectory);
+        if (directories != null) {
+            filterIndexPaths(tableDirs, directories, tableNames);
+        }
+    }
+
+    private static void filterIndexPaths(Map<Table, List<File>> tableDirs, File[] dirs, List<String> tableNames) {
+        for (String tableName : tableNames) {
+            List<File> indexDirs = tableDirs.get(new Table(tableName, false));
+            if (indexDirs == null) {
+                indexDirs = new ArrayList<>(dirs.length);
+            }
+            for (File dir : dirs) {
+                String dirName = dir.getName();
+                String partitionedPattern = ".*\\.partitioned\\." + tableName + "\\..+";
```

Comment:
can we re-use helpers of PartitionName here? like PartitionName.isPartition(dirName)..